snake.js

```diff
@@ -0,0 +1,190 @@
+const gameTimer = setInterval(draw,125);
+
+let canvas;
+let ctx;
+
+let tile = 32;
+let mapSize = [22, 16]
+let mapWidth = mapSize[0]*tile;
+let mapHeight = mapSize[1]*tile;
+
+let snake = [];
+
+function generateCoordinate (limit) {
+    return Math.floor(Math.random()*limit) * tile
+}
+
+function generateCoordinates () {
+    return {
+        x : generateCoordinate(mapSize[0]),
+        y : generateCoordinate(mapSize[1])
+    }
+}
+
+snake[0] = generateCoordinates();
+
+let berry = generateCoordinates();
+
+
+function draw(){
+    ctx = canvas.getContext("2d");
+    ctx.clearRect(0, 0, mapWidth+(tile*5), mapHeight);
+
+    tileMap();
+    
+    ctx.beginPath();
+
+    let score = (snake.length-1)*10;
+
+    menu(score);
+    
+    for(let i = 0; i< snake.length; i++){
+        ctx.fillStyle = 'green';
+        ctx.fillRect(snake[i].x, snake[i].y, tile, tile);
+    }
+
+    ctx.fillStyle = 'red';
+    ctx.fillRect(berry.x, berry.y, tile, tile)
+
+    let snakeXcoord = snake[0].x;
+    let snakeYcoord = snake[0].y;
+
+    const newCoords = move(snakeXcoord, snakeYcoord);
+    
+    if(newCoords[0] == berry.x && newCoords[1] == berry.y){
+        console.log(score);
+        changeBerryPosition(); //berry = generateCoordinates();
+    }else{
+        snake.pop();
+    }
+        
+    let tale = {
+        x : newCoords[0],
+        y : newCoords[1]
+    }
+
+    snake.unshift(tale);
+
+    if(gameOver(tale)){
+        console.log(true);
+        clearInterval(gameTimer);
+        
+        ctx.font = '75px times new roman';
+        ctx.fillStyle = 'red'
+       // ctx.strokeText();
+        ctx.fillText("Game Over", mapWidth/3, mapHeight/2.2);
+
+    }
+}
+
+function menu(score){
+    ctx.font = '30px times new roman';
+    ctx.strokeText("Your Score:", mapWidth+10, 50);
+    ctx.font = '40px times new roman';
+    let scoreSize = ctx.measureText(score) 
+//  console.log(text.width); 
+    ctx.strokeText(score, (mapWidth+tile*4.5)-scoreSize.width, tile*3);
+}
+
+function gameOver(tale){
+    if(snake[0].x < 0){
+        return true;
+    }
+    if(snake[0].y < 0){ 
+        return true;
+    }    
+    if(snake[0].x >= mapWidth){
+        return true;
+    }
+    if(snake[0].y >= mapHeight){
+        return true;
+    } else {
+        return false;
+    }
+    /*   for(let i = 0; i < tale.length; i++){
```

Comment:
This is right decision to detect collision between snake head and body. But it should be improved. Will do it at our next meeting.